src/main/java/Flash.java

```diff
@@ -0,0 +1,138 @@
+import java.io.File;
+import java.sql.Array;
+import java.util.Scanner;
+import java.util.List;
+import java.util.ArrayList;
+
+public class Flash {
```

Comment:
To improve abstraction, you can refactor the code by separating task-related logic from the main control flow. One class can handle the application's control logic (user interaction, commands) while another class can be responsible for managing tasks (adding, deleting, marking, saving, etc.).